src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -123,50 +123,37 @@ public virtual void FileWorkItems(SarifLog sarifLog)
                 return;
             }
 
-            for (int runIndex = 0; runIndex < sarifLog.Runs?.Count; ++runIndex)
+            IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+
+            PartitionFunction<string> partitionFunction = null;
+
+            switch (splittingStrategy)
             {
-                if (sarifLog.Runs[runIndex]?.Results?.Count > 0)
+                case SplittingStrategy.PerRun:
                 {
-                    IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+                    partitionFunction = (result) => result.ShouldBeFiled() ? "Include" : null;
+                    break;
+                }
+                case SplittingStrategy.PerResult:
+                {
+                    partitionFunction = (result) => result.ShouldBeFiled() ? Guid.NewGuid().ToString() : null;
+                    break;
+                }
+                default:
+                {
+                    throw new ArgumentOutOfRangeException($"SplittingStrategy: {splittingStrategy}");
+                }
+            }
 
-                    if (splittingStrategy != SplittingStrategy.PerRun)
-                    {
-                        SplittingVisitor visitor;
-                        switch (splittingStrategy)
-                        {
-                            case SplittingStrategy.PerRunPerRule:
-                                visitor = new PerRunPerRuleSplittingVisitor();
```

Comment:
>PerRunPerRuleSplittingVisitor [](start = 46, length = 29)

The splitting visitor pattern looks redundant with the partitioning pattern. I suggest we build this code out from the partitioning approach and drop that other code.

Reply:
Yes, the predicate-based partitioner is the way forward IMO.

---
In reply to: [394693923](https://github.com/microsoft/sarif-sdk/pull/1818#discussion_r394693923) [](ancestors = 394693923)